packages/devtools_shared/lib/src/deeplink/app_link_settings.dart

```diff
@@ -33,11 +34,31 @@ class AppLinkSettings {
   }
 
   /// Used when the the server can't retrieve app link settings.
-  static const empty = AppLinkSettings._('', false, <AndroidDeeplink>[]);
+  factory AppLinkSettings.fromErrorJson(String json) {
+    final jsonObject = jsonDecode(json) as Map;
+    final message = jsonObject[_kErrorKey]! as String;
+    return AppLinkSettings._(
+      '',
+      false,
+      <AndroidDeeplink>[],
+      message,
+    );
+  }
+
+  /// Used when the the server can't retrieve app link settings.
```

Comment:
can this dart doc be differentiated from the factor above so that a user of this API can know which to use for what cases?